src/app_service/tasks/marathon/mailserver/model.nim

```diff
@@ -63,26 +62,34 @@ proc newMailserverModel*(vptr: ByteAddress): MailserverModel =
   result.events = newMailserverEvents(vptr)
   result.nodes = initTable[string, MailserverStatus]()
   result.activeMailserver = ""
+  result.mailservers = @[]
 
 proc init*(self: MailserverModel) =
   trace "MailserverModel::init()"
-  let fleets =
-    if defined(windows) and defined(production):
+  
+  self.wakuVersion = status_settings.getWakuVersion()
+
+  let nodeConfig = status_settings.getNodeConfig()
+  if self.wakuVersion == 2:
+    # TODO:
+    # Instead of obtaining the waku2 fleet from fleet.json, expose a method in status-go that will
+    # return the list of store nodes. (The cluster config can ontain dns-discovery urls so it cannot be
+    # used to populate the list of mailservers)
+    let fleets = if defined(windows) and defined(production):
       "/../resources/fleets.json"
     else:
       "/../fleets.json"
+    let fleetConfig = readFile(joinPath(getAppDir(), fleets))
+    let fleet = newFleetModel(fleetConfig)
+    self.mailservers = toSeq(fleet.config.getMailservers(status_settings.getFleet(), true).values)
```

Comment:
nit: as in python when calling function with literal I would suggest to specify the name of the args like this:
```
     self.mailservers = toSeq(fleet.config.getMailservers(status_settings.getFleet(), isWakuV2=true).values)
```
All people reading your code will praise you for making their life easer ;)